src/components/Navigation.js

```diff
@@ -169,22 +213,196 @@ function Navigation() {
           <div className="d-flex flex-row align-items-center ms-auto navigation-search">
             <div className="d-flex flex-row align-items-center">
               <input
-                className="form-control me-2"
+                className="form-control me-2 bg-dark text-light navigation-search-input"
                 type="search"
-                placeholder="Search tx or address"
+                placeholder={`Search here`}
                 aria-label="Search"
                 ref={txSearchRef}
                 onKeyUp={handleKeyUp}
               />
-              <i className="fa fa-search pointer" onClick={search}></i>
             </div>
-            <Version />
           </div>
         </div>
+        <div className="mainnet-container hide-tabs">
+          <img
+            src={theme === 'dark' ? sun : moon}
+            alt="themeColorButton"
+            className="theme-color-btn"
+            onClick={() => dispatch(toggleTheme())}
+          />
+          <button className="mainnet-btn">
+            <GlobeMainnet
+              className={`${
+                theme === 'dark' ? 'dark-theme-logo' : 'light-theme-logo'
+              } theme-mainnet-btn`}
+            />
+            <span className="nav-title">Hathor Mainnet</span>
+          </button>
+        </div>
+        <div className="mobile-tabs">
+          {showSearchInput ? (
+            <input
+              type="search"
+              className={`form-control me-2 bg-dark text-light mobile-search-input ${
+                showSearchInput ? 'expanded' : ''
+              }`}
+              placeholder="Search..."
+              aria-label="Search"
+              ref={txSearchRef}
+              onKeyUp={handleKeyUp}
+              onBlur={() => setShowSearchInput(false)}
+              autoFocus
+            />
+          ) : (
+            <SearchIcon
+              fill={theme === 'dark' ? 'white' : 'black'}
+              onClick={toggleSearchInput}
+              className="mobile-search-icon"
+            />
+          )}
```

Comment:
Suggestion: Extract this into another component.

Just an idea, since `showSearchInput` can be moved to the component and the rest seems to work.
Also, this Navigation component would be more straightforward and easier to read.

Reply:
Sure!